test/comp.jl

```diff
@@ -97,3 +97,9 @@ end
     _test_monomials([x, y], 1:2, [y, x, y^2, x * y, x^2])
     _test_monomials([x, y], [0, 1, 3], [1, y, x, y^3, x*y^2, x^2*y, x^3])
 end
+
+@testset "Comparison with NaN" begin
+    @polyvar p
+    @test (NaN + p) != (NaN + p)
+    @test isequal(NaN + p, NaN + p)
```

Comment:
Can you also check that it's allocation-free ? Sometimes Julia decides to not specialize on function arguments so if you add `q = NaN + p` and then `@test 0 == @allocated q != q` and `@test 0 == @allocated isequal(q, q)`, it should make sure of that